common/dbIntrospection.ts

```diff
@@ -0,0 +1,82 @@
+import { Client } from "pg";
+import connectDBClient, { testDBClientConnection } from "./client";
+import { TableSchema } from "./interfaces";
+
+export default async function introspectDB(client: Client) {
+  if (!client) {
+    return;
+  }
+  const columnQuery = `
+      SELECT column_name, data_type, is_nullable, column_default
+      FROM information_schema.columns
+      WHERE table_schema = $1 AND table_name = $2
+      ORDER BY ordinal_position;
+      `;
+
+  const tableQuery = `
+    SELECT table_schema, table_name
+    FROM information_schema.tables
+    WHERE table_type = 'BASE TABLE'
+      AND table_schema NOT IN ('pg_catalog', 'information_schema');
+  `;
+
+  const foreignKeyQuery = `
+      SELECT 
+        kcu.column_name,
+        ccu.table_schema AS foreign_table_schema,
+        ccu.table_name AS foreign_table_name,
+        ccu.column_name AS foreign_column_name
+      FROM 
+        information_schema.table_constraints AS tc 
+        JOIN information_schema.key_column_usage AS kcu
+          ON tc.constraint_name = kcu.constraint_name
+          AND tc.table_schema = kcu.table_schema
+        JOIN information_schema.constraint_column_usage AS ccu
+          ON ccu.constraint_name = tc.constraint_name
+          AND ccu.table_schema = tc.table_schema
+      WHERE 
+        tc.constraint_type = 'FOREIGN KEY'
+        AND tc.table_schema = $1
+        AND tc.table_name = $2;
+      `;
+
+  client.connect();
```

Comment:
1. the client needs to be provided not connected?
2. this returns a promise, but you're not awaiting it. you should add a linter to catch hanging promises.